### DISCUSSION TITLE
Manual Install

### DISCUSSION THREAD

**Discussion by knapsea** [nasa/fprime#837] [Q&A] [answered]
How could I install F' Prime on a air-gapped computer, "pip install" and "git clone" isn't possible .
I have downloaded the .zip file of the git repository and can only work with it.

Can anyone suggest what does "pip install" do , so I could replicate its actions manually ?

Thanks
G

**Comment by SterlingPeet** [comment]
That command installs the Python package and any necessary dependencies from pypi.  You would need to go download the wheel files for each dependency manually, and then pip install those in reverse order.  Keep in mind that the dependency tree may be several layers deep, and all of the dependencies for a particular package will need to be installed before that package can be installed.

Sterling

Sent from my mobile.

On Jul 16, 2021, at 11:45 AM, gardnwater ***@***.***> wrote:

﻿

How could I install F' Prime on a air-gapped computer, "pip install" and "git clone" isn't possible .
I have downloaded the .zip file of the git repository and can only work with it.

Can anyone suggest what does "pip install" do , so I could replicate its actions manually ?

Thanks
G

**Reply by knapsea** [reply]
Thanks for your response.
This is the line for it "pip install --upgrade fprime-tools fprime-gds"
could I assume sources for fprime-tools and fprime-gds are  available in the git repo's .zip , would it work if could place them in python's path ?

Else If its easier , I do have another computer which is on the network, could I install FPrime on it , then copy any folders and configuration over ? 

Thanks
G

**Comment by SterlingPeet** [comment, marked as answer]
I don’t think it’s quite as easy as zipping up a folder.  However, I noticed that this article has a lot of good information about working with pip dependencies, so maybe you could gather up all your dependencies on a dev machine and then copy all the wheel files to install directly on your air-gapped machine.

https://www.activestate.com/resources/quick-reads/how-to-download-python-dependencies/

The process is ultimately scriptable, but you may not need to do it enough times to benefit from writing and debugging a script.

Sterling

Sent from my mobile.

On Jul 16, 2021, at 12:34 PM, gardnwater ***@***.***> wrote:

﻿

Thanks for your response.
This is the line for it "pip install --upgrade fprime-tools fprime-gds"
could I assume sources for fprime-tools and fprime-gds are available in the git repo's .zip , would it work if could place them in python's path ?

Else If its easier , I do have another computer which is on the network, could I install FPrime on it , then copy any folders and configuration over ?

Thanks
G

**Reply by knapsea** [reply]
Thanks.
I will try this , also found 
https://stackoverflow.com/questions/51830142/python-install-package-offline-with-dependencies-with-pip
